core/iwasm/fast-jit/fe/jit_emit_table.c

```diff
@@ -4,54 +4,315 @@
  */
 
 #include "jit_emit_table.h"
+#include "jit_emit_exception.h"
+#include "jit_emit_function.h"
+#include "wasm_runtime.h"
 #include "../jit_frontend.h"
 
+#if WASM_ENABLE_REF_TYPES != 0
 bool
 jit_compile_op_elem_drop(JitCompContext *cc, uint32 tbl_seg_idx)
 {
-    return false;
+    JitReg module, tbl_segs;
+
+    module = get_module_reg(cc->jit_frame);
+
+    tbl_segs = jit_cc_new_reg_ptr(cc);
+    GEN_INSN(LDPTR, tbl_segs, module,
+             NEW_CONST(I32, offsetof(WASMModule, table_segments)));
+
+    GEN_INSN(STI32, NEW_CONST(I32, true), tbl_segs,
+             NEW_CONST(I32, tbl_seg_idx * sizeof(WASMTableSeg)
+                                + offsetof(WASMTableSeg, is_dropped)));
+    return true;
 }
 
 bool
 jit_compile_op_table_get(JitCompContext *cc, uint32 tbl_idx)
 {
+    JitReg elem_idx, tbl_sz, tbl_data, elem_idx_long, offset, res;
+
+    POP_I32(elem_idx);
+
+    /* if (elem_idx >= tbl_sz) goto exception; */
+    tbl_sz = get_table_cur_size_reg(cc->jit_frame, tbl_idx);
+    GEN_INSN(CMP, cc->cmp_reg, elem_idx, tbl_sz);
+    if (!jit_emit_exception(cc, EXCE_OUT_OF_BOUNDS_TABLE_ACCESS, JIT_OP_BGEU,
+                            cc->cmp_reg, NULL))
+        goto fail;
+
+    elem_idx_long = jit_cc_new_reg_I64(cc);
+    GEN_INSN(I32TOI64, elem_idx_long, elem_idx);
+
+    offset = jit_cc_new_reg_I64(cc);
+    GEN_INSN(MUL, offset, elem_idx_long, NEW_CONST(I64, sizeof(uint32)));
```

Comment:
- it is wired that we didn't lock `RAX` and `RDX` in `int_mul` https://github.com/bytecodealliance/wasm-micro-runtime/blob/6e4b2dbc76dd9973d95edd6fe29f172f04d0c5fa/core/iwasm/fast-jit/fe/jit_emit_numberic.c#L438.
- in codegen, we will use `shl` to replace `mul` in some scenarios. https://github.com/bytecodealliance/wasm-micro-runtime/blob/6e4b2dbc76dd9973d95edd6fe29f172f04d0c5fa/core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp#L2274

Reply:
Have a check with jit_codegen_x86_64.cpp, we only use: IMUL r32, r/m32, IMUL r64, r/m64, IMUL r32, r/m32, imm32, IMUL r64, r/m64, imm32. They really don't need to lock eax/edx, see: https://www.felixcloutier.com/x86/imul
`IMUL r/m32` and `IMUL r/m64` requires locking eax/edx.

An issue is that `IMUL r64, r/m64, imm32` only supports imm32, there seems to be a bug when translating `MUL, r64_0, r64_1, imm64`: https://github.com/bytecodealliance/wasm-micro-runtime/blob/dev/fast_jit/core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp#L2619
The imm64 will be truncated into imm32. We should add check by comparing to `INT32_MIN/INT32_MAX`, if in the range, use `IMUL r64, r/m64, imm32`, otherwise, move imm64 to rsi, and use `IMUL r64, r/m64`.

For data == 2^n, we only handle data == 1/2/4/8, not including 16/32.